### PR TITLE
[Behat][Domain] Minor refactor domain contexts

### PR DESCRIPTION
**Pull request by GSadee**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Related tickets | 
| License         | MIT

### CODE REVIEW

File: src/Sylius/Behat/Context/Domain/OrderContext.php [hunk 1]
```diff
@@ -105,7 +99,7 @@ public function iDeleteTheOrder(OrderInterface $order)
     }
 
     /**
-     * @Then /^([^"]+) should not exist in the registry$/
+     * @Then /^(this order) should not exist in the registry$/
      */
     public function orderShouldNotExistInTheRegistry(OrderInterface $order)
```
Comment (michalmarcinkowski) [inline src/Sylius/Behat/Context/Domain/OrderContext.php]: We shouldn't use repository in this method also.

**Comment by michalmarcinkowski** [comment]
The class should be renamed to match the corresponding context in UI namespace `ManagingOrdersContext`.

**Comment by pjedrzejewski** [comment]
Thanks Grzesiu!

File: src/Sylius/Behat/Context/Domain/ManagingProductsContext.php [hunk 2]
```diff
@@ -66,31 +66,26 @@ public function __construct(
...
-    {
-        $this->sharedStorage->set('product_variant_id', $productVariant->getId());
-        $this->productVariantRepository->remove($productVariant);
-    }
-
-    /**
      * @When /^I try to delete the ("[^"]+" variant of product "[^"]+")$/
      */
-    public function iTryToDeleteTheVariantOfProduct(ProductVariantInterface $productVariant)
+    public function iDeleteTheVariantOfProduct(ProductVariantInterface $productVariant)
```
Comment (michalmarcinkowski) [inline src/Sylius/Behat/Context/Domain/ManagingProductsContext.php]: There is a reason for keeping `I delete` and `I try to delete` methods separate. For domain implementation we mute the exception in step `I try to delete` so we can assert it occurred, but for step `I delete` we shouldn't do that and let it be thrown. This change should be reverted.

File: src/Sylius/Behat/Context/Domain/ManagingProductsContext.php [hunk 3]
```diff
@@ -117,39 +112,28 @@ public function productVariantShouldNotExistInTheProductCatalog()
...
+     * @Then this variant should still exist in the product catalog
      */
-    public function productVariantShouldExistInTheProductCatalog(ProductVariantInterface $productVariant)
+    public function productVariantShouldExistInTheProductCatalog()
     {
-        $productVariant = $this->productVariantRepository->find($productVariant->getId());
+        $productVariantId = $this->sharedStorage->get('product_variant_id');
+        $productVariant = $this->productVariantRepository->find($productVariantId);
 
         Assert::notNull($productVariant);
```
Comment (michalmarcinkowski) [inline src/Sylius/Behat/Context/Domain/ManagingProductsContext.php]: How about the following?
```
/**
 * @Then /^(this variant) should still exist in the product catalog$/
 */
public function productVariantShouldExistInTheProductCatalog(ProductVariantInterface $productVariant = null)
{
    Assert::notNull($productVariant);
} 
```

File: src/Sylius/Behat/Context/Domain/ManagingPromotionCouponsContext.php [hunk 4]
```diff
@@ -45,31 +45,20 @@ public function __construct(SharedStorageInterface $sharedStorage, CouponReposit
...
+     * @When /^I delete ("[^"]+" coupon) related to (this promotion)$/
      * @When /^I try to delete ("[^"]+" coupon) related to (this promotion)$/
      */
     public function iTryToDeleteCoupon(CouponInterface $coupon, PromotionInterface $promotion)
     {
         try {
             $promotion->removeCoupon($coupon);
             $this->couponRepository->remove($coupon);
-
-            throw new \Exception(sprintf('Coupon "%s" has been removed, but it should not.', $coupon->getCode()));
```
Comment (michalmarcinkowski) [inline src/Sylius/Behat/Context/Domain/ManagingPromotionCouponsContext.php]: It should not be removed (see my comment above).

**Pull request merged**